packages/plugins/Traderxyz/src/SNSAdaptor/PostPreview.tsx

```diff
@@ -0,0 +1,339 @@
+        }
+
+        if (orderStatus === 6) {
+            // Cancelled
+            showSnackbar(t.order_cancelled(), { variant: 'error' })
+        }
+
+        if (orderStatus === 0) {
+            // Cancelled
+            showSnackbar(t.order_invalid(), { variant: 'error' })
+        }
+
+        if (orderStatus === 1) {
+            // Invalid
+            showSnackbar(t.order_invalid_maker(), { variant: 'error' })
+        }
+
+        if (orderStatus === 2) {
+            // Invalid
+            showSnackbar(t.order_invalid_taker(), { variant: 'error' })
+        }
+    }
+
+    const s = info.assetsInfo.preview_info
+
+    const nftList = s.nftMediaUrls
+    const previewImages = nftList.map((item: nftData) => {
+        return (
+            <>
+                <Grid className={classes.previewBoxInnerGridContainerItem} padding={1}>
+                    <img
+                        className={classes.previewBoxInnerGridContainerItemImg}
+                        alt={item.nft_name}
+                        src={item.image_preview_url}
+                    />
+                </Grid>
+            </>
+        )
+    })
+
+    const chipTitle = s.receivingSymbol.amount + ' ' + s.receivingSymbol.symbol
+
+    let labelString = ''
+
+    if (nftList.length > 0) {
+        labelString = `Buy ${nftList[0]?.nft_name} (#${nftList[0]?.nft_id})`
+
+        if (nftList.length > 1) {
+            labelString += ` and ${nftList[1]?.nft_name} (#${nftList[1]?.nft_id})`
+        }
+
+        if (nftList.length > 2) {
+            labelString += ' and Others'
+        }
+
+        labelString += ' for ' + chipTitle
+    }
+    usePluginWrapper(true)
+    return (
+        <Box className={classes.previewBoxOuter} padding={1}>
+            <Box className={classes.previewBoxInner}>
+                <Grid className={classes.previewBoxInnerGridContainer} padding={0}>
+                    {previewImages}
+                </Grid>
+                <Chip className={classes.previewBoxInnerGridContainerChip} label={chipTitle} />
+            </Box>
+            <Grid container direction="column" justifyContent="center" alignItems="flex-start">
+                <Grid item>
+                    <Typography paddingTop={1} variant="subtitle1" className={classes.previewBoxInnerGridFooterTitle1}>
+                        {labelString}
+                    </Typography>
+                </Grid>
+            </Grid>
+            <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }} className={classes.buttonWrapper}>
+                <Grid item xs={6}>
+                    <Button className={classes.button} onClick={sharePost} fullWidth size="small" variant="contained">
+                        Share
+                    </Button>
+                </Grid>
+                <Grid item xs={6}>
+                    <Button className={classes.button} onClick={signOrder} fullWidth size="small" variant="contained">
```

Comment:
Recommend to use `<ActionButtonPromise />`.

Reply:
done.